homeassistant/components/freebox/number.py

```diff
@@ -0,0 +1,130 @@
+"""Support for Freebox Delta, Revolution and Mini 4K."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+
+from freebox_api.exceptions import InsufficientPermissionsError
+
+from homeassistant.components.number import (
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .router import FreeboxRouter
+
+
+@dataclass
+class FreeboxNumberSetValueMixin:
+    """Mixin for required keys."""
+
+    async_set_value: Callable[[FreeboxRouter, int], Awaitable]
+
+
+@dataclass
+class FreeboxNumberEntityDescription(
+    NumberEntityDescription, FreeboxNumberSetValueMixin
+):
+    """Class describing Freebox number entities."""
+
+
+LCD_NUMBER_DESCRIPTIONS: tuple[FreeboxNumberEntityDescription, ...] = (
+    FreeboxNumberEntityDescription(
+        key="lcd_brightness",
+        name="LCD Brightness",
+        icon="mdi:monitor",
+        entity_category=EntityCategory.CONFIG,
+        native_min_value=0,
+        native_max_value=100,
+        mode=NumberMode.SLIDER,
+        async_set_value=lambda router, value: router.set_brightness(brightness=value),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the switch."""
+    router: FreeboxRouter = hass.data[DOMAIN][entry.unique_id]
+    entities = [
+        FreeboxLCDNumber(router, entity_description)
+        for entity_description in LCD_NUMBER_DESCRIPTIONS
+    ]
+    async_add_entities(entities, True)
+
+
+class FreeboxNumber(NumberEntity):
+    """Representation of a Freebox number entity."""
+
+    entity_description: FreeboxNumberEntityDescription
+
+    def __init__(
+        self,
+        router: FreeboxRouter,
+        entity_description: FreeboxNumberEntityDescription,
+    ) -> None:
+        """Initialize the switch."""
+        self.entity_description = entity_description
+        self._attr_name = f"Freebox {entity_description.name}"
```

Comment:
Instead of adding/prefixing the device name, you could also set `_attr_has_entity_name = True` as a class variable and remove this line.